spec/p2p/reactor/reactor.md

```diff
@@ -0,0 +1,228 @@
+# Reactor API
+
+A component has to implement the [`p2p.Reactor`][reactor-interface] interface
+in order to use communication services provided by the p2p layer.
+This interface is currently the main source of documentation for a reactor.
+
+The goal of this document is to specify the behaviour of the p2p communication
+layer when interacting with a reactor.
+So while the [`Reactor`][reactor-interface] interface declares the methods
+invoked and determines what the p2p layer expects from a reactor,
+this documentation focuses on the behaviour that a reactor implementation
+should expect from the p2p layer.
+
+This specification is accompanied by the [`reactor.qnt`](./reactor.qnt) file,
+a more comprehensive model of the reactor's operation written in
+[Quint][quint-repo], an executable specification language.
+The methods declared in the [`Reactor`][reactor-interface] interface are
+modeled in Quint, in the form of `pure def` methods, providing some examples of
+how they should be implemented.
+The behaviour of the p2p layer when interacting with a reactor, by invoking the
+interface methods, is modeled in the form of state transitions, or `action`s in
+the Quint nomenclature.
+
+## Overview
+
+The following _grammar_ is a simplified representation of the expected sequence of calls
+from the p2p layer to a reactor.
+Note that the grammar represents events referring to a _single reactor_, while
+the p2p layer supports the execution of multiple reactors.
+
+While it is useful to provide an overview of the operation of a reactor,
+grammars have some limitations in terms of the behaviour they can express.
+For instance, the following grammar only represents the management of _a single peer_,
+namely of a peer with a given ID which can connect, disconnect, and reconnect
+multiple times to the node.
```

Comment:
Can we say that the Quint spec actually is capturing the sequences of calls more precisely?

Reply:
Mentioned in the paragraph above.